apps/vscode-extension/src/commands/extract/styling/helpers/utils.helper.ts

```diff
@@ -0,0 +1,202 @@
+import { Position, Selection, TextDocument, TextEditorEdit, TextLine, workspace } from 'vscode';
+import { basename } from 'node:path';
+import { EOL } from 'node:os';
+import {regExp} from './regex.helper';
+import * as vscode from 'vscode';
+
+/**
+ * Sanitaze the selector
+ *
+ * @param selector to clean
+ * @returns the cleaned selector
+ */
+function cleanSelector(selector: string): string {
+  return selector.replaceAll(regExp.trimLeft, '')
+    .replaceAll(regExp.trimRight, '')
+    .replaceAll(regExp.sanitizer, '-');
+}
+
+/**
+ * Generate the proposed variable name based of the path of selectors.
+ * Also make sure to not include any duplicated selectors or any forbidden words from the config `forbiddenWords`
+ *
+ * @param selectors to used to generate the name
+ * @param fileName name of the file to append to the generated name
+ * @returns the final proposed name
+ */
+function generateScssVarName(selectors: string[], fileName: string): string {
+  let name = '';
+  const forbiddenWords = vscode.workspace.getConfiguration('otter.extract.styling').get<string>('forbiddenWords');
+  const wordsNotToReuseSet = new Set<string>([...forbiddenWords ?? [], '']);
+  const prefixToAdd = workspace.getConfiguration('otter.extract.styling').get<string>('prefix');
+
+  selectors = [...([prefixToAdd ?? '']), fileName, ...selectors];
+  selectors.forEach((element) => {
+    const elementWords = element.split('-');
+    let distinctSelector = '';
+    elementWords.forEach((word) => {
+      if (!wordsNotToReuseSet.has(word)) {
+        wordsNotToReuseSet.add(word);
+        distinctSelector = distinctSelector.concat(word, '-');
+      }
+    });
+    name = name.concat(distinctSelector);
+  });
+  return name;
+}
+
+/**
+ * Go through the tree of selectors and keep only the path to the selected property
+ *
+ * / ! \ Important to have this point in mind to understand this function :
```

Comment:
Is it missing comments or are the parameters the important information to have in mind?